src/box/memtx_space.c

```diff
@@ -87,6 +89,20 @@ memtx_space_update_bsize(struct space *space, struct tuple *old_tuple,
 	memtx_space->bsize += new_bsize - old_bsize;
 }
 
+void
+memtx_space_update_compressed_tuples(struct space *space,
+				     struct tuple *old_tuple,
+				     struct tuple *new_tuple)
+{
+	struct memtx_space *memtx_space = (struct memtx_space *)space;
+	if (old_tuple != NULL && tuple_is_compressed(old_tuple))
+		memtx_space->compressed_tuples--;
+	if (new_tuple != NULL && tuple_is_compressed(new_tuple))
+		memtx_space->compressed_tuples++;
+	if (memtx_space->compressed_tuples == 0)
```

Comment:
You must not call `memtx_space_update_indexes_vtab` every type `compressed_tuples` is zero. You must call it only if it _becomes_ zero. Otherwise it will have an extra overhead for spaces which have never used compression.

Reply:
... Or becomes nonzero.

Reply:
Yes moreover we don't need to update vtab in case when we change format